src/mc.rs

```diff
@@ -136,72 +140,285 @@ const SUBPEL_FILTERS: [[[i32; SUBPEL_FILTER_SIZE]; 16]; 6] = [
   ]
 ];
 
+// gets an index that can be mapped to a function for a pair of filter modes
+const fn get_2d_mode_idx(mode_x: FilterMode, mode_y: FilterMode) -> usize {
+  mode_x as usize + 4 * (mode_y as usize)
+}
+
+type PutFn =
+  unsafe extern fn(*mut u8, isize, *const u8, isize, i32, i32, i32, i32);
+type PutHBDFn = unsafe extern fn(
+  *mut u16,
+  isize,
+  *const u16,
+  isize,
+  i32,
+  i32,
+  i32,
+  i32,
+  i32
+);
+
+union PutFnUnion {
```

Comment:
this could be an enum

Reply:
Wouldn't that use another byte if the structs are passed around?

Reply:
We can check the compiler output, ideally it could be not needed though ^^; (But to try I'll need to have time and that will happen only in few days)

Reply:
You cannot have both None, I assume.

Reply:
I don't think I understand what you are asking.

Reply:
I can play with that myself later. this set is taking already enough time :)